test/decl/protocol/conforms/nscoding.swift

```diff
@@ -1,7 +1,7 @@
-// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -parse-as-library -swift-version 4 %s -verify
-// RUN: not %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -parse-as-library -swift-version 4 %s -disable-nskeyedarchiver-diagnostics 2>&1 | %FileCheck -check-prefix CHECK-NO-DIAGS %s
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -parse-as-library -swift-version 4 %s -target x86_64-apple-macosx10.11 -verify
+// RUN: not %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -parse-as-library -swift-version 4 %s -target x86_64-apple-macosx10.11 -disable-nskeyedarchiver-diagnostics 2>&1 | %FileCheck -check-prefix CHECK-NO-DIAGS %s
 
-// RUN: not %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -parse-as-library -swift-version 4 %s -dump-ast 2> %t.ast
+// RUN: not %target-swift-frontend(mock-sdk: %clang-importer-sdk) -typecheck -parse-as-library -swift-version 4 %s -target x86_64-apple-macosx10.11 -dump-ast 2> %t.ast
```

Comment:
This won't work on non-macOS platforms (which it is important to continue testing for everything else).

Reply:
I thought that, but convinced myself that regardless of what platform we build on, we can still target macOS (so long as we aren't running code, and subject to the ObjC interop requirement already in place). That's why I did a full test run, which includes all of the simulators, and they passed. Do you foresee a further problem that wouldn't show up there?

Also, I don't feel like we're done here, because availability isn't getting transferred to nested classes in the way I expected.

Reply:
That's not correct; our test bots just don't test any cross-compile-to-iOS-only configurations. We have some inside Apple.

Reply:
I think I have a Radar for the latter problem, but I wasn't planning to fix it for the 4.0 branch…

Reply:
I'll split out the test case.